src/models/subscription.ts

```diff
@@ -194,6 +194,14 @@ export default class Subscription extends Sequelize.Model {
 	// This is a workaround to fix a long standing bug in sequelize for JSON data types
 	// https://github.com/sequelize/sequelize/issues/4387
 	async updateSyncState(updatedState: RepoSyncStateObject): Promise<Subscription> {
+
+		if (!updatedState?.repos) {
```

Comment:
This was pulled out from the discovery function, brought it here so we only have to call update once from discovery.... however, looking at it now i am concerned the and update with no repos may be a legitimate case?
@mboudreau @KAbakumovAtl

Reply:
Yes sure, if there is no repos we don't need to run backfill queue.

If we pull it out from the discovery queue code we need to add a unit test for this case for this class tests

Reply:
also, you're just checking to see if the repos object exist, it could be possible it's an empty object.  Maybe try something like `if(!Object.keys(updatedState?.repos || {}).length) {`

Reply:
This will only get called once from discovery, and handles with/without repos, so discovery is not directly updating status anymore(win). It returns early if nothing to update so no unnecessary checks are run,